src/main/scala/diplomaticobjectmodel/model/OMRTLModule.scala

```diff
@@ -31,22 +35,79 @@ trait Asynchronous extends Synchronicity
 case class OMRTLReset(
   activeEdge: Option[OMSignalAssertionLevel],
   clock: String, // This will always be the name of the clock signal on the to p-level module
-  synchronicity: Option[Synchronicity]
-)
+  synchronicity: Option[Synchronicity],
+  _types: Seq[String] = Seq("OMClockReset", "OMRTLComponent", "OMCompoundType")
+) extends OMRTLComponent
 
+case class OMReset(
+  description: Option[String],
+  name: String,
+  activeEdge: Option[OMSignalAssertionLevel],
+  clockName: String, // This will always be the name of the clock signal on the top-level module
+  synchronicity: Option[Synchronicity],
+  uid: Int,
+  _types: Seq[String] = Seq("OMReset", "OMSignal", "OMRTLComponent", "OMCompoundType")
+) extends OMSignal
+
+// Status
+trait OMStatus extends OMSignal {
+  def description: Option[String]
+  def name: String
+  uid: Int
```

Comment:
I think you mixed the syntax for case classes with traits here.

Reply:
Not sure why you are calling this out here in particular... there is already quite a mix of those in this file.

Reply:
No, I'm specifically referring to the fact that this particular trait uses both `def` and non-`def`. In every other thing in this file, all the Traits only use `def` and all the case classes just use constructor arguments. I don't actually understand what this syntax means, and I originally thought it would be a syntax error. Can you explain what this syntax means, since I haven't seen it before? Traits in Scala aren't allowed to define a constructor, so I don't really know what this specific `uid` line means.